pkg/util/exec/fake_exec.go

```
@@ -143,3 +145,69 @@ func (fake *FakeExitError) Exited() bool {
+
+func (fake *FakeExec) expectFakeCmd(command interface{}, fcmd *FakeCmd) {
+	expectedArgv := commandAsStringSlice(command)
+	fake.CommandScript = append(fake.CommandScript,
+		func(cmd string, args ...string) Cmd {
+			InitFakeCmd(fcmd, cmd, args...)
+			if !reflect.DeepEqual(expectedArgv, fcmd.Argv) {
+				panic(fmt.Sprintf("Wrong Exec: expected %v got %v", expectedArgv, fcmd.Argv))
+			}
+			return fcmd
+		},
+	)
+}
+
+// ExpectCombinedOutput "predicts" a call to fake.Command(...).CombinedOutput() with the
+// given command, and provides the result of that call.
+// command is either a string (command line) or a []string (argv) indicating the expected command.
+// output and err will be returned as the result of CombinedOutput() on the command.
+// output can be either a []byte or a string (which will be converted to a []byte).
+func (fake *FakeExec) ExpectCombinedOutput(command interface{}, output interface{}, err error) {
+	fake.expectFakeCmd(command, &FakeCmd{
+		CombinedOutputScript: []FakeCombinedOutputAction{
+			func() ([]byte, error) { return outputAsBytes(output), err },
+		},
+	})
+}
+
+// ExpectRun "predicts" a call to fake.Command(...).Run() with the
+// given command, and provides the result of that call.
+// command is either a string (command line) or a []string (argv) indicating the expected command.
+// stdout, stderr, and err will be returned as the result of Run() on the command.
+// stdout and stderr can be either a []byte or a string (which will be converted to a []byte).
+func (fake *FakeExec) ExpectRun(command interface{}, stdout interface{}, stderr interface{}, err error) {
+	fake.expectFakeCmd(command, &FakeCmd{
+		RunScript: []FakeRunAction{
+			func() ([]byte, []byte, error) { return outputAsBytes(stdout), outputAsBytes(stderr), err },
+		},
+	})
+}
+
+// AssertExpectedCommands ensures that all of the commands added to fake via ExpectCombinedOutput and ExpectRun were actually executed
+func (fake *FakeExec) AssertExpectedCommands() {
```

yeah, why not just return it, and let the caller panic or t.Fatalf or whatever